### PR TITLE
RavenDB-14006 compare exchange includes

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

**Comment by ppekrol** [comment]
TODO:

- support for queries

File: src/Raven.Client/Documents/Session/AbstractDocumentQuery.Includes.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+            if (includes.DocumentsToInclude != null)
+            {
+                foreach (var doc in includes.DocumentsToInclude)
+                {
+                    DocumentIncludes.Add(doc);
+                }
+            }
+
+            IncludeCounters(includes.Alias, includes.CountersToIncludeBySourcePath);
```
Comment (ayende) [inline src/Raven.Client/Documents/Session/AbstractDocumentQuery.Includes.cs:55]: Can't `CountersToIncludeBySourcePath` be null?

**Review by ayende — APPROVED** [review]
Note that we are missing JS API for this, though.
I think that is relevant for counters, timeseries, etc. 
We need a way to call that from JS, because that is the escape hatch fro complexity

File: src/Raven.Client/Documents/Session/AbstractDocumentQuery.Includes.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Client/Documents/Session/AbstractDocumentQuery.Includes.cs:55]: I just moved the old code, but the method 'IncludeCounters' checks that internally.

**Comment by ppekrol** [comment]
I've added a TODO for this, this can be merged I think

**Pull request merged**